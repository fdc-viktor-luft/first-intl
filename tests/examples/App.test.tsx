import '../setupTests-intl'; // this is something you could include into your regular test setup
import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import React from 'react';
import { App } from './App';

describe('App', () => {
    it('renders the App without translations, but all necessary information', () => {
        // see the "__snapshots__" directory within this directory to the see the rendered output
        expect(render(<App />).container.firstChild).toMatchSnapshot();
    });
});
