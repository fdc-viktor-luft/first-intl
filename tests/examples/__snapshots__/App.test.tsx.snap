// Vitest Snapshot v1

exports[`App > renders the App without translations, but all necessary information 1`] = `
<div>
  <header>
    <img
      alt="__('header.logo.alt')"
      src="/path/to/img"
    />
  </header>
  <main>
    <button>
      __('account.remove')
    </button>
    <p>
      __({ id: 'account.remove.info', values: {email: 'my.example@mail.com'} })
    </p>
    __({ id: 'account.tac.text', values: {br: &lt;br /&gt;, info: "__({ id: 'account.tac.info' })", tac: &lt;a href="/path/to/tac"&gt;__({ id: 'account.tac' })&lt;/a&gt;} })
  </main>
  <footer>
    <a
      href="/path/to/home"
    >
      __({ id: 'footer.navigation.home' })
    </a>
    <img
      alt="__('footer.logo.alt')"
      src="/path/to/img-footer"
    />
  </footer>
</div>
`;
